ui/src/authors/containers/AssignAllDifferentProfileActionContainer.jsx

```diff
@@ -0,0 +1,37 @@
+import { connect } from 'react-redux';
+
+import AssignDifferentProfileAction from '../components/AssignDifferentProfileAction';
+import {
+  assignDifferentProfileUnclaimedPapers,
+  assignDifferentProfileClaimedPapers,
+} from '../../actions/authors';
+
+const stateToProps = (state) => ({
+  disabled: state.authors.get('publicationSelection').size === 0,
+  currentUserId: state.user.getIn(['data', 'recid']),
+  claimingUnclaimedPapersDisabled:
+    state.authors.get('publicationSelectionUnclaimed').size === 0,
+  claimingClaimedPapersDisabled:
+    state.authors.get('publicationSelectionClaimed').size === 0,
+});
+
+const dispatchToProps = (dispatch) => ({
+  onAssign({
+    from,
+    to,
+    claimingUnclaimedPapersDisabled,
+    claimingClaimedPapersDisabled,
+  }) {
+    if (!claimingUnclaimedPapersDisabled) {
```

Comment:
could we have two different dispatches?

`onAssignWithoutUnclaimed` and `onAssignWithoutClaimed` for example.

Reply:
but we need to decide somewhere which one we activate and we have only one button for that in the Action, so we need to know which function we need to use `onClick`